lib/agent0/agent0/hyperdrive/policies/lpandarb.py

```diff
@@ -0,0 +1,642 @@
+"""Agent policy for LP trading that also arbitrage on the fixed rate."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import TYPE_CHECKING
+
+from fixedpointmath import FixedPoint
+
+from agent0.hyperdrive.state import HyperdriveActionType, HyperdriveMarketAction
+from elfpy.types import MarketType, Trade
+
+from .hyperdrive_policy import HyperdrivePolicy
+
+if TYPE_CHECKING:
+    from ethpy.hyperdrive import HyperdriveInterface
+    from numpy.random._generator import Generator as NumpyGenerator
+
+    from agent0.hyperdrive.state import HyperdriveWallet
+
+import time
+
+# pylint: disable=too-many-parameters, too-many-local-variables
+
+# constants
+fp0 = FixedPoint(0)
+fp1 = FixedPoint(1)
+fp2 = FixedPoint(2)
+fp12 = FixedPoint(12)
+fp_seconds_in_year = FixedPoint(365 * 24 * 60 * 60)
+tolerance = 1e-18
+MAX_ITER = 10
+MIN_TRADE_AMOUNT = FixedPoint(scaled_value=20)
+
+
+# functions
+def calc_bond_reserves(
```

Comment:
we have `bonds_given_shares_and_rate` in the interface. Which I believe should be the same as your `calc_bond_reserves`. The math looks different, and I think the api math is correct because includes zeta adjustment. Can you look into this & resolve it so we're only using one (preferably a correct version in the api)

Reply:
adjusted for zeta adjustment, otherwise they're the same